### ISSUE TITLE
dg crash when compiled in debug mode

### ISSUE THREAD

**Issue by nesfield** [mchalupa/dg#212]
Slicing the following code around either __assert_fail or printf when dg is compiled in debug mode causes dg to crash:
```
#include <stdio.h>
#include <stdlib.h>
#include <assert.h>

void sayHello();
void sayGoodbye();

void sayHello() {
  printf("hello world\n");
}

void sayGoodbye() {
  printf("Goodbye world!\n");
  assert(0);
}

int main() {
  void (*sayHelloPtr)() = sayHello;
  void (*sayGoodbyePtr)() = sayGoodbye;
  int i = rand();
  if(i == 1)
    (*sayHelloPtr)();
  else
    (*sayGoodbyePtr)();
}
```

Compiled with llvm6:
`clang -g -c -emit-llvm func_ptr.c -o func_ptr.bc`

Sliced with (dg compiled in debug mode):
`dg/tools/llvm-slicer -c printf -annotate=slicer func_ptr.bc`
`dg/tools/llvm-slicer -c __assert_fail -annotate=slicer func_ptr.bc`

If we slice with dg compiled in release mode around printf and run the slice:
`dg_release/tools/llvm-slicer -c printf -annotate=slicer func_ptr.bc`
`lli func_ptr.sliced`
The output is usually an infinite sequence of "Goodbye world!" printouts (instead of one, and then a failed assert or exiting the program).

**Comment by nesfield** [comment]
It triggers an abort here: dg/src/llvm/analysis/PointsTo/PointsTo.h:99

This is the output:
```
../dg/tools/llvm-slicer -c printf -annotate=slicer func_ptr.bc
Pointer Subgraph is broken!
This happend after building this function called via pointer: sayGoodbye
Invalid number of edges:
PSNodeType::NOOP with ID 39
  - operands: []
(Non-root node has no predecessors)
Unreachable node:
PSNodeType::CALL_RETURN with ID 29
  - operands: [37 PSNodeType::CALL_RETURN]
Unreachable node:
PSNodeType::CALL_RETURN with ID 37
  - operands: []
Unreachable node:
PSNodeType::NOOP with ID 39
  - operands: []
../dg/tools/llvm-slicer(_ZN4llvm3sys15PrintStackTraceERNS_11raw_ostreamE+0x33)[0x9b3017]
../dg/tools/llvm-slicer[0x9b30aa]
../dg/tools/llvm-slicer(_ZN4llvm3sys17RunSignalHandlersEv+0x99)[0x9b1293]
../dg/tools/llvm-slicer[0x9b288c]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x11390)[0x7f1a9697c390]
/lib/x86_64-linux-gnu/libc.so.6(gsignal+0x38)[0x7f1a95d2e428]
/lib/x86_64-linux-gnu/libc.so.6(abort+0x16a)[0x7f1a95d3002a]
../dg/tools/llvm-slicer(_ZN2dg23LLVMPointerAnalysisImplINS_8analysis3pta23PointsToFlowInsensitiveEE19functionPointerCallEPNS2_6PSNodeES6_+0x333)[0x89adef]
dg/src/libPTA.so(_ZN2dg8analysis3pta15PointerAnalysis11processNodeEPNS1_6PSNodeE+0x893)[0x7f1a970602b1]
../dg/tools/llvm-slicer(_ZN2dg8analysis3pta15PointerAnalysis3runEv+0x163)[0x846a61]
../dg/tools/llvm-slicer(_ZN2dg19LLVMPointerAnalysis3runINS_8analysis3pta23PointsToFlowInsensitiveEEEvv+0x67)[0x85d365]
../dg/tools/llvm-slicer(_ZN6Slicer7buildDGEv+0x8d)[0x85276d]
../dg/tools/llvm-slicer(main+0x8c2)[0x84118b]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf0)[0x7f1a95d19830]
../dg/tools/llvm-slicer(_start+0x29)[0x83e6e9]
Stack dump:
0.	Program arguments: ../dg/tools/llvm-slicer -c printf -annotate=slicer func_ptr.bc 
Aborted (core dumped)
```

This is the stack:
```
#0  0x00007ffff6044428 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
#1  0x00007ffff604602a in __GI_abort () at abort.c:89
#2  0x000000000068691f in dg::LLVMPointerAnalysisImpl<dg::analysis::pta::PointsToFlowInsensitive>::functionPointerCall (this=0x7fffffffde30, callsite=0x13c98d0, called=0x13c8580)
    at dg/src/llvm/analysis/PointsTo/PointsTo.h:99
#3  0x00007ffff73762b1 in dg::analysis::pta::PointerAnalysis::processNode (this=0x7fffffffde30, node=0x13c98d0) at dg/src/analysis/PointsTo/PointerAnalysis.cpp:378
#4  0x0000000000632591 in dg::analysis::pta::PointerAnalysis::run (this=0x7fffffffde30) at dg/src/analysis/PointsTo/PointerAnalysis.h:138
#5  0x0000000000648e95 in dg::LLVMPointerAnalysis::run<dg::analysis::pta::PointsToFlowInsensitive> (this=0x13cbb20) at dg/src/llvm/analysis/PointsTo/PointsTo.h:213
#6  0x000000000063e29d in Slicer::buildDG (this=0x7fffffffe060) at dg/tools/llvm-slicer.cpp:430
#7  0x000000000062ccbb in main (argc=5, argv=0x7fffffffea88) at dg/tools/llvm-slicer.cpp:797
```

**Comment by mchalupa** [comment]
I fixed #222 that should fix this bug. Does the fix work for you?

**Comment by mchalupa** [comment]
IMHO this should be fixed. Please, reopen if needed.
